tests/check_for_valid_readme.py

```diff
@@ -0,0 +1,53 @@
+"""Test script to check README.rst files for syntax errors."""
+import argparse
+import sys
+from pathlib import Path
+
+import readme_renderer.rst
+
+
+def is_valid_rst(path):
+    """Checks if RST can be rendered on PyPI."""
+    with open(path) as f:
+        markup = f.read()
+    if readme_renderer.rst.render(markup) is None:
+        return False
+    return True
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Checks README.rst file in path for syntax errors."
+    )
+    parser.add_argument(
+        "paths", nargs="+", help="paths containing a README.rst to test"
+    )
+    parser.add_argument("-v", "--verbose", action="store_true")
+    return parser.parse_args()
+
+
+if __name__ == "__main__":
+    args = parse_args()
+
+    error = False
+    all_readmes_found = True
+
+    for path in [Path(path) for path in args.paths]:
+        readme = path / "README.rst"
+        if not readme.exists():
+            all_readmes_found = False
+            print("✗ No README.rst in", str(path))
+            continue
+        if not is_valid_rst(readme):
+            error = True
+            print("✗ RST syntax errors in", readme)
+            continue
+        if args.verbose:
+            print("✓", readme)
+
+    if error:
+        sys.exit(1)
+    if all_readmes_found:
+        print("All clear.")
+    else:
+        print("No errors found but not all packages have a README.rst")
```

Comment:
Any reason not to treat this as an error and exit as above? I would assume we'd want a README.rst for all packages.

Reply:
The package does not technically need to have a README.rst. The file could be named differently, it could be an markdown or plain text file, or the description could be embedded inline, instead of in a separate file. `testutil` does not have a README and seems to be marked for deletion (#374 5.), so I felt this behavior might be desirable.

If someone adds a README.rst it will get checked, and if they do not include a README.rst we just assume that the package is still WIP and/or will not be uploaded to PyPI. If there is another README file the warning basically indicates departure from implicit convention.

Reply:
We can just force everyone to put the readme in README.rst. If we want to allow README.md, then this tool will need to support it too.

If we want to exclude packages, I think we can add the exclusion list to eachdist.ini (but I'm not sure if it has support for configured exclusions yet).

Reply:
Exclusions do not seem to be supported, so I added a README to the `testutil` package for now.